tools/test/build_api/build_api_test.py

```diff
@@ -238,5 +240,41 @@ def test_build_library_no_app_config(self, mock_prepare_toolchain, mock_exists,
         self.assertEqual(args[1]['app_config'], None,
                          "prepare_toolchain was called with an incorrect app_config")
 
+    @patch('tools.build_api.Resources')
+    @patch('tools.build_api.mkdir')
+    @patch('os.path.exists')
+    @patch('tools.build_api.prepare_toolchain')
+    def test_merge_region_no_fit(self, mock_prepare_toolchain, mock_exists, _, __):
+        """
+        Test that merge region fails as expected when part size overflows region size.
+        """
+        with patch("tools.build_api.intelhex_offset") as _intelhex_offset:
+            max_addr = 87444
+            # create a dummy hex file with above max_addr
+            _intelhex_offset.return_value = IntelHex({1:2, max_addr:0})
+
+            # create application and post-application regions and merge.
+            region_application = Region("application", 10000, 86000, True, "random.hex")
+            region_post_application = Region("postapplication", 100000, 90000, False, None)
+
+            notify = MockNotifier()
+            mock_prepare_toolchain.config = MagicMock(
+                                has_regions=True, name=None, lib_config_data=None)
+            region_list = [region_application, region_post_application]
+            region_list = list(region_list)
+            toolchain = prepare_toolchain(self.src_paths, self.build_path, self.target,
+                                          self.toolchain_name, notify=notify)
```

Comment:
You don't need a toolchain object at all, so you can delete this line and the line above